src/Resources/Resources.Test/ScenarioTests/TagTests.ps1

```diff
@@ -0,0 +1,266 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Utility method to test CreeteOrUpdate for Tags within tracked resources and subscription.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParams($resourceId)
+{
+    # Setup
+    $expected = @{"key1"="value1"; "key2"="value2";}
+
+    try
+    {
+        # Test
+        $res = New-AzTag -ResourceId $resourceId -Tag $expected
+        [hashtable]$actual = $res.Properties.TagsProperty
+
+        # Assert
+        Assert-AreEqual $expected.Count $actual.Count
+        Assert-AreEqual $expected.key1 $actual.key1
```

Comment:
compare value as well .. value should be check with case sensitivity, key is case insensitive

Reply:
Added a function to compare simple hash tables: with key case insensitive, and value case sensitive.